src/gpio.rs

```diff
@@ -143,49 +143,52 @@ macro_rules! gpio {
             }
 
             impl<MODE> OutputPin for $PXx<Output<MODE>> {
-                fn set_high(&mut self) {
+                type Error = ();
```

Comment:
I think I read somewhere that the prefered error type for a non-failable pin is `!` rather than `()`. Though the never type isn't fully implemented iirc